xgcm/grid.py

```diff
@@ -758,6 +765,122 @@ def max(
             keep_coords,
         )
 
+    def transform(
+        self,
+        da,
+        target,
+        target_data=None,
+        target_dim=None,
+        method="linear",
+        mask_edges=False,
+    ):
+        """Transform data along a 1D axis
+
+        Parameters
+        ----------
+        da : xr.Dataarray
+            Input data
+        target : {np.array, xr.Dataarray}
+            Target data for transformation. Dependin on the method is interpreted as cell center (method='linear') or cell bounds (method='conservative).
+            When passed as numpy array the resulting dimension is named according to `target_data`, if provided as xr.Dataarray naming is inferred from dimensions.
```

Comment:
Can you be clearer here? Does it have the same name as `target_data`? If so, just say that.